From 59dab9411e5830894d69f745dbecbb61877ee466 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Fri, 1 Jul 2016 15:37:43 +0300
Subject: [PATCH 136/278] drivers/usb: Make Axxia USB Use Coherent Memory for
 DMA

Even with 'dma-coherent' in the device tree USB node, the XHCI
driver creates devices without copying the 'archdata' structure,
where the dma_coherent field is defined.  To handle this, set
the dma_coherent field in all devices that are created by the
XHCI driver (match 'xhci-hcd' in the device name).

Signed-off-by: John Jacques <john.jacques@intel.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 drivers/usb/dwc3/dwc3-axxia.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/usb/dwc3/dwc3-axxia.c b/drivers/usb/dwc3/dwc3-axxia.c
index 59321b8..a0de833 100644
--- a/drivers/usb/dwc3/dwc3-axxia.c
+++ b/drivers/usb/dwc3/dwc3-axxia.c
@@ -79,6 +79,20 @@ static int axxia_dwc3_remove(struct platform_device *pdev)
 	return 0;
 }
 
+void
+arch_setup_pdev_archdata(struct platform_device *pdev)
+{
+
+	printk("setup_pdev_archdata %s\n", pdev->name);
+
+	if (strncmp(pdev->name, "xhci-hcd", strlen("xhci-hcd")) == 0) {
+		printk("setting dma_coherent\n");
+		pdev->dev.archdata.dma_coherent = 1;
+
+	}
+
+}
+
 static const struct of_device_id adwc3_of_match[] = {
 	{ .compatible = "intel,axxia-dwc3", },
 	{},
-- 
1.8.5.2.233.g932f7e4

